code/modules/capture_the_flag/ctf_panel.dm

```diff
@@ -0,0 +1,51 @@
+GLOBAL_DATUM_INIT(ctf_panel, /datum/ctf_panel, new())
+
+/datum/ctf_panel
+
+/datum/ctf_panel/ui_state(mob/user)
+	return GLOB.observer_state
+
+/datum/ctf_panel/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "CTFPanel")
+		ui.open()
+
+/datum/ctf_panel/ui_data(mob/user)
+	var/list/data = list()
+	data["teams"] = list()
+	data["enabled"] = ""
+	for(var/obj/machinery/capture_the_flag/team in GLOB.machines)
+		var/list/this = list()
+		this["name"] = team
+		this["color"] = team.team
+		this["score"] = team.points + team.control_points
+		this["team_size"] = team.team_members.len
+		this["refs"] += "[REF(team)]"	
+		data["teams"] += list(this)
+		if(team.ctf_enabled)
+			data["enabled"] = "CTF is currently running!"
+		else
+			data["enabled"] = "CTF needs 4 players to start, currently [team.people_who_want_to_play.len]/4 have signed up!"
+	return data
+
+
+/datum/ctf_panel/ui_act(action, params, datum/tgui/ui)
+	.= ..()
+	if(.)
+		return
+	var/mob/user = ui.user
+
+	switch(action)
+		if("jump")
+			var/obj/machinery/capture_the_flag/ctf_spawner = locate(params["refs"]) in GLOB.machines 
```

Comment:
```suggestion
			var/obj/machinery/capture_the_flag/ctf_spawner = locate(params["refs"])
```

You don't need this because the ref will already make a direct search.